README.md

```diff
@@ -17,11 +25,22 @@ ________________________________________________________________________________
     - More Details on Federated Learning : **https://federated.withgoogle.com/**
 
 - Uses MNIST dataset of different writing styles considering them as data of various client's.
+- Train model on federated data; predicting and customizing model implementation for accurate results.
 
+## Enhanced Data Security and Use Cases:
 
-- Train model on federated data; predicting and customizing model implementation for accurate results.
-_________________________________________________________________________________________________________________________________________________________________
-_________________________________________________________________________________________________________________________________________________________________
+Federated learning works without storing user data in the cloud, but that is not all. There has been a **Secure Agression Potocol** put in place which used crytographic techniques so a server(cloud) can decrypt **average update** only when 100s or 1000s of users/devices have participated - no individual data cannot be accessed befor averaging. Federated Averaging was designed so that the coordinating server only needs the average update, allowing Secure Aggregation to be used; however, the protocol is general and can be applied to other problems as well.
+
+
+Applying Federated Learning requires machine learning practitioners to adopt new tools and a new way of thinking: model development, training, and evaluation with no direct access to or labeling of raw data, with communication cost as a limiting factor.
+
+#### Use Cases:
+
+- Gboard query suggestions. 
+- Photo rankings based on what kinds of photos people look at, share, or delete.
```

Comment:
Please add relevent and specific examples, as photo ranking could be implemented using collaborative filtering and ranking.